test/cleanup.go

```diff
@@ -22,16 +22,36 @@ package test
 import (
 	"os"
 	"os/signal"
+	"sync"
 )
 
+// cleanFuncs stores all the functions to cleanup
+type cleanFuncs struct {
+	once sync.Once
+	s    *sync.Map
+}
+
+func init() {
+	cf = &cleanFuncs{s: &sync.Map{}, once: sync.Once{}}
+}
+
+var cf *cleanFuncs
```

Comment:
```suggestion
// cleanFuncs stores all the functions to cleanup
var cf struct {
	m sync.Map
	o sync.Once
}
```

Why complicate things?